pkg/connectors/highlevel_client_test.go

```diff
@@ -0,0 +1,259 @@
+		pause := false
+		running := true // running true or false is irrelevant
+		callDeployConnector(t, pause, running, configOnline, configLocal)
+	})
+}
+
+func Test_DeployMultipleConnector_Ok(t *testing.T) {
+	mockBaseClient := NewMockBaseClient(t)
+	client := newHighLevelClient(mockBaseClient, 2)
+
+	lock := &sync.Mutex{}
+	received := map[string]interface{}{}
+
+	connectorRequests := []CreateConnectorRequest{}
+	for i := 1; i <= 5; i++ {
+		connectorName := fmt.Sprintf("test%d", i)
+		request := CreateConnectorRequest{
+			ConnectorRequest: ConnectorRequest{connectorName},
+		}
+		mockBaseClient.On("GetConnector", request.ConnectorRequest).
+			Return(ConnectorResponse{Name: connectorName}, nil).Once()
+		mockBaseClient.On("GetConnectorConfig", request.ConnectorRequest).
+			Return(GetConnectorConfigResponse{}, nil).Once()
+
+		mockBaseClient.On("UpdateConnector", request).Return(ConnectorResponse{}, nil).Run(func(args mock.Arguments) {
+			req := args.Get(0).(CreateConnectorRequest)
+			lock.Lock()
+			defer lock.Unlock()
+			received[req.Name] = true
+		}).Once()
+		mockBaseClient.On("GetConnectorConfig", request.ConnectorRequest).
+			Return(GetConnectorConfigResponse{Config: map[string]interface{}{"name": connectorName}}, nil).Once()
+
+		connectorRequests = append(connectorRequests, request)
+	}
+
+	err := client.DeployMultipleConnector(connectorRequests)
+
+	assert.Equal(t, map[string]interface{}{"test1": true, "test2": true, "test3": true, "test4": true, "test5": true}, received)
+	assert.NoError(t, err)
+}
+
+func Test_DeployMultipleConnector_Error(t *testing.T) {
+	mockBaseClient := NewMockBaseClient(t)
+	client := newHighLevelClient(mockBaseClient, 2)
+
+	mockBaseClient.On("GetConnector", mock.Anything).Times(5).Return(ConnectorResponse{}, errors.New("random error"))
+
+	err := client.DeployMultipleConnector([]CreateConnectorRequest{
+		{ConnectorRequest: ConnectorRequest{Name: "test1"}},
+		{ConnectorRequest: ConnectorRequest{Name: "test2"}},
+		{ConnectorRequest: ConnectorRequest{Name: "test3"}},
+		{ConnectorRequest: ConnectorRequest{Name: "test4"}},
+		{ConnectorRequest: ConnectorRequest{Name: "test5"}},
+	})
+
+	assert.Error(t, err)
+}
+
+func newHighLevelClient(mockBaseClient *MockBaseClient, maxParallelRequest int) *highLevelClient {
+	logger, _ := logrusTest.NewNullLogger()
+
+	return &highLevelClient{
+		client:             mockBaseClient,
+		maxParallelRequest: maxParallelRequest,
+		logger:             logrus.NewEntry(logger),
+	}
+}
+
+// callDeployConnector is a helper function to test DeployConnector
+// when we ask to pause the connector we expect:
+// - get connector and config to perform comparison
+// - if running: status call + pause/resume
+// - if NOT running: status call only
+// - update the connector and compare config until they are identical
+func callDeployConnector(t *testing.T, pauseBeforeDeploy, isRunning bool, configOnline, configLocal map[string]interface{}) {
+	mockBaseClient := NewMockBaseClient(t)
```

Comment:
`t.Helper()` ?

This would help debug when one of the test case using helper function fails.